src/circuit_builder.rs

```diff
@@ -10,111 +10,64 @@ use crate::commitment_scheme::kzg10::PublicParameters;
 use crate::constraint_system::StandardComposer;
 use crate::error::Error;
 use crate::proof_system::{Proof, ProverKey, VerifierKey};
+#[cfg(feature = "canon")]
+use canonical::Canon;
+#[cfg(feature = "canon")]
+use canonical_derive::Canon;
 use dusk_bls12_381::BlsScalar;
-use dusk_bytes::Serializable;
 use dusk_jubjub::{JubJubAffine, JubJubScalar};
 
-const BLS_SCALAR: u8 = 1;
-const JUBJUB_SCALAR: u8 = 2;
-const JUBJUB_AFFINE: u8 = 3;
+#[derive(Default, Debug, Clone)]
+#[cfg_attr(feature = "canon", derive(Canon))]
+/// Structure that represents a PLONK Circuit Public Input
+/// structure converted into it's &[BlsScalar] repr.
+pub struct PublicInputValue(pub(crate) Vec<BlsScalar>);
```

Comment:
The name suggests its a single public input when its in fact a collection

Reply:
And in fact it is a Public Input. But expressed on it's BlsScalar form. And as you know. Since we have 3 types that are Public Inputs and one of them (`AffinePoint`) gets represented as 2 BlsScalar instead of one. 

Therefore the best way we've found with @ZER0 for now is to return a `Vec<BlsScalar>` which will be 1 item for BlsScalar and JubJubScalar and 2 for AffinePoint.

Reply:
Before it was an enum. Do we have a gain to convert from enum to this `Vec` structure? Also, if we know the size to be either 1 or 2, any of these simpler structures will also do:

* `(BlsScalar, Option<BlsScalar>)`
* `[Option<BlsScalar>; 2]`
* `[BlsScalar; 2]`

Reply:
But Vec seemed much simpler than that. The only one that I might adopt is the Array. But is more missleading since makes it look like it's always 2 arrays instead of one.

Reply:
As alternative we could use const generic also here, you can play with this if you want and see how the API looks like.